current.go

```diff
@@ -72,12 +73,22 @@ func (w *CurrentWeatherData) SetLang(lang string) error {
 // CurrentByName will provide the current weather with the provided
 // location name.
 func (w *CurrentWeatherData) CurrentByName(location string) error {
-	response, err := http.Get(fmt.Sprintf(fmt.Sprintf(baseURL, "q=%s&units=%s&lang=%s"), location, DataUnits[w.Unit], w.Lang))
-	if err != nil {
-		return err
+	var err error
+	var response *http.Response
+	switch {
+	case strings.Contains(location, " "):
```

Comment:
I guess it'd make sense to just escape each time.  I didn't want to do it
if there was no spaces to require it though it'd probably be the same
amount of operations if I do.

On Mon, Jan 19, 2015 at 4:10 AM, Fábio Gomes <notifications@github.com>
wrote:

> In current.go
> <https://github.com/briandowns/openweathermap/pull/12#discussion-diff-23153795>
> :
>
> > @@ -72,12 +73,22 @@ func (w *CurrentWeatherData) SetLang(lang string) error {
> >  // CurrentByName will provide the current weather with the provided
> >  // location name.
> >  func (w *CurrentWeatherData) CurrentByName(location string) error {
> > -	response, err := http.Get(fmt.Sprintf(fmt.Sprintf(baseURL, "q=%s&units=%s&lang=%s"), location, DataUnits[w.Unit], w.Lang))
> > -	if err != nil {
> > -		return err
> > +	var err error
> > +	var response *http.Response
> > +	switch {
> > +	case strings.Contains(location, " "):
>
> Why is this necessary? Why not just escape always?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/briandowns/openweathermap/pull/12/files#r23153795>.
>



-- 
Brian